### ISSUE TITLE
Damage Absorption via Barrier

### ISSUE THREAD

**Issue by Ilandria** [Krappa322/arcdps_healing_stats#30]
I dug through the options and I couldn't find any way to add Barrier damage absorption to the log... I know absorption doesn't always show up in combat text (it seems like only absorbing self-inflicted damage does), but ArcDPS has a "barrier" stat in their Self Stats window somehow.

Even if it's not possible to distinguish absorption source by spell, would it be possible to have an amalgamated "barrier" bar in the healing skills window via using whatever data ArcDPS is displaying in their Self Stats window?

I mostly ask as I play Scourge and depending on my build and the context of the fight anywhere from around 25% to 100% of my total "healing" is done via barrier (especially during more intense moments), so it would help to see that.

All of the above said, please point me in the correct direction if I'm just blind!

**Comment by Krappa322** [comment]
Indeed barrier generation is not shown in the in-game window today. It is logged to the log however and elite insights will parse it correctly to show generated barrier, as far as I'm aware.

Barrier absorption is a much harder topic and I don't think that's handled much at all. The bigger issue here is that since healing stats gets the events from other players, and the arcdps area stats gets the events from just the game server, the timestamps don't line up perfectly. So it's very hard to correlate and tell you for example how much barrier you had (and who generated that barrier) for a single damage event. It might be possible if healing stats were to log all events (right now it just logs healing events, but it wouldn't be a huge change to log all of them), and elite insights was able to parse the whole log just using healing stats as the source of truth. In any case this is not a development I'm willing to undertake at this moment, but I would of course be happy to help and give pointers if anybody else wants to try to achieve it.

So in short, yes your request is very valid and healing stats does not handle this currently, but unfortunately this is too big for me to try to pursue at this moment

**Comment by Ilandria** [comment]
Yeah no worries, thanks for the context and info!
